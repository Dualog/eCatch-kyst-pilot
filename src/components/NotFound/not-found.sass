.not-found

  img
    pointer-events: none
    max-width: 50vh