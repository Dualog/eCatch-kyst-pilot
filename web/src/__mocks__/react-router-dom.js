import React from 'react'
import {BrowserRouter} from "react-router-dom"

module.exports = {
  BrowserRouter: ({children}) => <div>{children}</div>,
  Route: ({children}) => <div>{children}</div>,
  Link: ({children, to}) => <a href="mocked" to={to}>{children}</a>,
  withRouter: () => Component => props => <BrowserRouter><Component {...props}/></BrowserRouter>,
  Redirect: ({to}) => <div to={to}/>,
  Switch: ({children}) => <div>{children}</div>
}