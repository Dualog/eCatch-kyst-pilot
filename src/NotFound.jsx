import React from 'react'
import {Link} from "react-router-dom"
import {withNamespaces} from 'react-i18next'
import {routes} from './lib/router'


const NotFound = ({t}) =>
  <div className="not-found">
    <p>{t('page.notfound')}</p>
    <Link to={routes.ROOT}>{t('page.goBackToMain')}</Link>
  </div>

export default withNamespaces("common")(NotFound)