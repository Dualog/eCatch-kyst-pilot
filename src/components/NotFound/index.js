import React from 'react'
import {Link} from "react-router-dom"
import {withTranslation} from 'react-i18next'

import {Button, Typography} from '@material-ui/core'

import {routes} from '../../lib/router'
import empty from "../../assets/empty_state.svg"

import "./not-found.sass"

import {useDimensions} from '../../hooks'
import Centered from '../Centered'

const NotFound = ({t}) => {

  const {height, width} = useDimensions()

  return (
    <Centered>
      <Typography variant="h3">404</Typography>
      <Typography variant="h5">
        {t('not-found.text')}
      </Typography>
      <img
        alt={t("not-found.img-alt")}
        src={empty}
        style={{margin: `${height/10}px 0`}}
        width={width/2}
      />
      <Button
        color="secondary"
        component={Link}
        size="large"
        to={routes.ROOT}
        variant="contained"
      >
        {t('not-found.back-to-main')}
      </Button>
    </Centered>
  )
}

export default withTranslation("common")(NotFound)