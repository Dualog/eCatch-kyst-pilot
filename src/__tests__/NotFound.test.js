import NotFound from "../NotFound"
import {Link} from "react-router-dom"

describe("NotFound component", () => {
  const wrapper = mount(<NotFound/>)

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("Paragraph translated", () => {
    expect(wrapper.find("p").props("children").children).toBe("page.notfound")
  })

  it("Link translated", () => {
    expect(wrapper.find(Link).props("children").children).toBe("page.goBackToMain")
  })
})