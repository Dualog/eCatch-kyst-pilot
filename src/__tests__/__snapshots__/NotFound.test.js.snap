// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component renders correctly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {},
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "t": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="not-found"
        >
          <p>
            page.notfound
          </p>
          <a
            href="mocked"
            to="/"
          >
            page.goBackToMain
          </a>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              page.notfound
            </p>,
            <Link
              to="/"
            >
              page.goBackToMain
            </Link>,
          ],
          "className": "not-found",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <p>
              page.notfound
            </p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "page.notfound",
            },
            "ref": null,
            "rendered": Array [
              "page.notfound",
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "page.goBackToMain",
              "to": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": <a
                href="mocked"
                to="/"
              >
                page.goBackToMain
              </a>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "page.goBackToMain",
                "href": "mocked",
                "to": "/",
              },
              "ref": null,
              "rendered": Array [
                "page.goBackToMain",
              ],
              "type": "a",
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {},
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "t": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": <div
            class="not-found"
          >
            <p>
              page.notfound
            </p>
            <a
              href="mocked"
              to="/"
            >
              page.goBackToMain
            </a>
          </div>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                page.notfound
              </p>,
              <Link
                to="/"
              >
                page.goBackToMain
              </Link>,
            ],
            "className": "not-found",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <p>
                page.notfound
              </p>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "page.notfound",
              },
              "ref": null,
              "rendered": Array [
                "page.notfound",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "page.goBackToMain",
                "to": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": <a
                  href="mocked"
                  to="/"
                >
                  page.goBackToMain
                </a>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "page.goBackToMain",
                  "href": "mocked",
                  "to": "/",
                },
                "ref": null,
                "rendered": Array [
                  "page.goBackToMain",
                ],
                "type": "a",
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
